core-bundle/src/Migration/Version500/BasicEntitiesMigration.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration\Version500;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Contao\StringUtil;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class BasicEntitiesMigration extends AbstractMigration
+{
+    private static array $targets = [
```

Comment:
Every extension will have to copy this. Can we come up with a better solution?

Reply:
Not sure if that's the way to go. We'll have the same problem in the future anyway. Especially with input/output encoding. But also recent PRs like e.g. https://github.com/contao/contao/pull/4396 provide migrations which contain hardcoded targets and we should somehow provide a common way. Not sure if always creating an abstract makes sense or if we should introduce some general concept? @contao/developers

Reply:
I thought about iterating over the DCA, but modifying database columns with such a general string replacement seems problematic to me. So I think every extension should have to hard-code the columns in a migration in order to be safe. Also not replacing some `[&]`s would not be so bad IMO, but replacing it in a column that stores binary data or so would be very bad.

Reply:
See #4505